### PR TITLE
Bug fix: await on send message

### PR DESCRIPTION
**Pull request by sujeebant**

### CODE REVIEW

File: src/library/masq.js [hunk 1]
```diff
@@ -317,17 +317,19 @@ class Masq {
...
           const userAppKey = Buffer.from(json.key, 'hex')
-          this.appsDBs[dbName].authorize(userAppKey, (err) => {
-            if (err) throw err
-            sendWriteAccessGranted(peer)
-            this.sw.close()
-            return resolve()
-          })
+          try {
+            await this.appsDBs[dbName].authorizeAsync(userAppKey)
+          } catch (err) {
```
Comment (G-Ray) [inline src/library/masq.js:322]: not necessary to catch right ?

Comment (G-Ray) [reply src/library/masq.js:322]: or if we need to catch we may close the connection

**Comment by G-Ray** [comment]
It think you already pushed theses changes to https://github.com/QwantResearch/masq-app/pull/27/ ?

File: src/library/masq.js (hunk 1 above)
Comment (sujeebant) [reply src/library/masq.js:322]: Yes it is not necessary but we should not rethrow the same error, we should specify an error type with a MasqError. It is done in #27

**Comment by sujeebant** [comment]
> It think you already pushed theses changes to #27 ?

Yes but it will be removed when this PR is merged (done to make it work and pass CI tests before this PR is merged)

File: src/library/masq.js (hunk 1 above)
Comment (G-Ray) [reply src/library/masq.js:322]: Ok but maybe we need to close the connecttion at least ?

Comment (sujeebant) [reply src/library/masq.js:322]: I guess we do... but I think there is at least one other error case in which we do not close the connection, at the execution of _checkProfile, if it throws we should close the connection and rethrow ?

**Review by G-Ray — APPROVED** [review]

**Pull request merged**